sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -656,6 +667,81 @@ __SYCL_TYPE(dynamic_work_group_memory) dynamic_work_group_memory
 
   void __init(decoratedPtr Ptr) { this->WorkGroupMem.__init(Ptr); }
 #endif
+
+#ifdef __SYCL_DEVICE_ONLY__
+  [[maybe_unused]] unsigned char
+      Padding[sizeof(detail::dynamic_work_group_memory_base)];
+#endif
+};
+
+template <typename DataT, int Dimensions = 1>
+class __SYCL_SPECIAL_CLASS
+__SYCL_TYPE(dynamic_local_accessor) dynamic_local_accessor
+#ifndef __SYCL_DEVICE_ONLY__
+    : public detail::dynamic_local_accessor_base
+#endif
+{
+public:
+  static_assert(Dimensions > 0 && Dimensions <= 3);
+
+  // Frontend requires special types to have a default constructor in order to
+  // have a uniform way of initializing an object of special type to then call
+  // the __init method on it. This is purely an implementation detail and not
+  // part of the spec.
+  // TODO: Revisit this once https://github.com/intel/llvm/issues/16061 is
+  // closed.
+  dynamic_local_accessor() = default;
+
+  /// Constructs a new dynamic_local_accessor object.
+  /// @param Graph The graph associated with this object.
+  /// @param AllocationSize The size of the local accessor.
+  /// @param PropList List of properties for the underlying accessor.
+  dynamic_local_accessor(
+      [[maybe_unused]] experimental::command_graph<graph_state::modifiable>
+          Graph,
+      [[maybe_unused]] range<Dimensions> AllocationSize,
+      [[maybe_unused]] const property_list &PropList = {})
+#ifndef __SYCL_DEVICE_ONLY__
+      : detail::dynamic_local_accessor_base(
+            detail::convertToArrayOfN<3, 1>(AllocationSize), Dimensions,
+            sizeof(DataT), PropList)
+#endif
+  {
+  }
+
+  local_accessor<DataT, Dimensions> get() const {
```

Comment:
In the spec PR I don't think we say this method is const, so that is something to update in the spec

Reply:
I have now updated the spec to mark the get() methods as const